src/python/tests/gff3/test_id_allocator.py

```diff
@@ -0,0 +1,250 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.gff3.id_allocator` module."""
+
+from contextlib import nullcontext as does_not_raise
+from difflib import unified_diff
+import filecmp
+from pathlib import Path
+from typing import ContextManager, List, Optional
+
+import pytest
+
+from BCBio import GFF
+from Bio.SeqRecord import SeqRecord
+from ensembl.io.genomio.gff3.id_allocator import StableIDAllocator, InvalidStableID
+from pytest import raises
+
+
+def _read_record(in_gff: Path) -> SeqRecord:
+    """Get one clean record from a GFF3 file."""
+    with in_gff.open("r") as in_gff_fh:
+        for record in GFF.parse(in_gff_fh):
+            new_record = SeqRecord(seq=record.seq, id=record.id)
+            new_record.features = record.features
+            return new_record
+
+    raise ValueError(f"No feature loaded from {in_gff}")
+
```

Comment:
Do we want to make this a fixture as we have done for the automatic load of JSON files?

Reply:
Yeah I plan to do that in a new PR